src/solvers/dgsem_p4est/dg_2d.jl

```diff
@@ -547,19 +557,25 @@ end
     # The nonconservative flux is scaled by a factor of 0.5 based on
     # the interpretation of global SBP operators coupled discontinuously via
     # central fluxes/SATs
-    noncons = nonconservative_flux(u_ll, u_rr, normal_direction, equations)
+    noncons_primary = nonconservative_flux(u_ll, u_rr, normal_direction, equations)
+    noncons_secondary = nonconservative_flux(u_rr, u_ll, normal_direction, equations)
 
-    flux_plus_noncons = flux + 0.5f0 * noncons
+    flux_plus_noncons_primary = flux + 0.5f0 * noncons_primary
+    flux_plus_noncons_secondary = flux + 0.5f0 * noncons_secondary
```

Comment:
Why do we use `Float32` here for `0.5`?

Reply:
We do the same from your merged 3D PR (https://github.com/trixi-framework/Trixi.jl/blob/main/src/solvers/dgsem_p4est/dg_3d.jl#L642-L643). 

Julia always promotes to the higher precision (IIRC). So, e.g., when `flux` and `noncons_primary` are `Float64` the complete result will get promoted properly. However, when `flux` and `noncons_primary` are `Float32` then the type get preserved. If we were to use `0.5` then this `Float64` would promote up if `flux` and `noncons_primary` are `Float32`.

Reply:
Thanks for the explanation!

> We do the same from your merged 3D PR

I know. I just adopted what was there already, but I didn't know the reason.